lib/trie/print.go

```diff
@@ -25,38 +26,49 @@ func (t *Trie) String() string {
 func (t *Trie) string(tree gotree.Tree, curr node, idx int) {
 	switch c := curr.(type) {
 	case *branch:
-		hasher := newHasher(false)
-		defer hasher.returnToPool()
-		c.encoding, _ = hasher.encode(c)
+		buffer := encodingBufferPool.Get().(*bytes.Buffer)
+		buffer.Reset()
+
+		const parallel = false
+		_ = encodeBranch(c, buffer, parallel)
+		c.encoding = buffer.Bytes()
+
 		var bstr string
 		if len(c.encoding) > 1024 {
 			bstr = fmt.Sprintf("idx=%d %s hash=%x gen=%d", idx, c.String(), common.MustBlake2bHash(c.encoding), c.generation)
 		} else {
 			bstr = fmt.Sprintf("idx=%d %s encode=%x gen=%d", idx, c.String(), c.encoding, c.generation)
 		}
+
+		encodingBufferPool.Put(buffer)
+
 		sub := tree.Add(bstr)
 		for i, child := range c.children {
 			if child != nil {
 				t.string(sub, child, i)
 			}
 		}
 	case *leaf:
-		hasher := newHasher(false)
-		defer hasher.returnToPool()
-		c.encoding, _ = hasher.encode(c)
+		buffer := encodingBufferPool.Get().(*bytes.Buffer)
+		buffer.Reset()
+
+		_ = encodeLeaf(c, buffer)
+
+		c.encodingMu.Lock()
+		defer c.encodingMu.Unlock()
+		c.encoding = buffer.Bytes()
+
 		var bstr string
 		if len(c.encoding) > 1024 {
 			bstr = fmt.Sprintf("idx=%d %s hash=%x gen=%d", idx, c.String(), common.MustBlake2bHash(c.encoding), c.generation)
 		} else {
 			bstr = fmt.Sprintf("idx=%d %s encode=%x gen=%d", idx, c.String(), c.encoding, c.generation)
 		}
+
+		encodingBufferPool.Put(buffer)
+
 		tree.Add(bstr)
 	default:
 		return
 	}
 }
-
-// Print prints the trie through pre-order traversal
-func (t *Trie) Print() {
-	fmt.Println(t.String())
-}
```

Comment:
did you move this? or does the trie already have a `String()` func somewhere else?

Reply:
Yeah it does have a `String()` method (:eyes: see the body of `Print()` :wink:). And I removed it since it's unused.